### PR TITLE
Service/Redux for Schedule/Reschedule Observations

### PR DESCRIPTION
**Pull request by karthikbtf**
- added service/types/redux to schedule/reschedule observations
- refactored existing code to enable reuse of async reducers

### CODE REVIEW

**Review by constanzauanini — APPROVED** [review]

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -97,7 +97,7 @@ jobs:
           path: cypress/screenshots
 
       - name: build
-        run: yarn build
+        run: NODE_OPTIONS="--max-old-space-size=2048" yarn build
```
Comment (karthikbtf) [inline .github/workflows/workflow.yml:100]: @constanzauanini I added this for now to see if it helps (see [stackoverflow](https://stackoverflow.com/questions/53230823/fatal-error-ineffective-mark-compacts-near-heap-limit-allocation-failed-javas)) we could still debug the issue later

Comment (constanzauanini) [reply .github/workflows/workflow.yml:100]: I almost at the same time created a PR to increase memory

Comment (constanzauanini) [reply .github/workflows/workflow.yml:100]: https://github.com/terraware/terraware-web/pull/1721

Comment (karthikbtf) [reply .github/workflows/workflow.yml:100]: Saw that. Should we just do this for github workflows (where the problem exists) versus for everyone that runs the build?

**Pull request merged**